CONTRIBUTING.md

```diff
@@ -0,0 +1,125 @@
+# Contributing
+
+Thanks for helping with SlimeVR! This document explains what to know when
+contributing.
+
+We are most active on [discord](https://discord.gg/SlimeVR), come chat with us if you
+have any questions or want to hang out! 😎
+
+
+## Setting up the development environment
+
+We use platformio as our build tooling and rely on git for version control. There are
+instructions to set all of this up [here](https://docs.slimevr.dev/firmware/setup-and-install.html).
+
+
+## Using third-party code
+
+Often you will want to use or reference third party code. You should not just
+copy-paste code from random libraries! There are several reasons for this:
+
+- It is not clear when/where the code originated from.
+- The original code's license needs to be kept intact.
+- It removes the ability to update our code when the original updates.
+- It makes it harder for us to maintain the code, because we don't what has changed
+  from the original!
+
+Instead, we will want to do the following:
+1. Check that the dependency has a permissive license
+2. (Optional) Fork the dependency if it needs any modification
+1. Add the dependency via platformio
+
+
+### Check the License
+
+The SlimeVR firmware is under the MIT License and also is intended to be permissively
+licensed. This means that we will not accept contributions that include code from any
+viral software licenses. If you do not know what that means, here is a breakdown of some
+common software licenses:
+
+> Fine for use:
+> * MIT
+> * Apache 2.0
+> * BSD
+>
+> Will not be accepted:
+> * GPLv3 or v2
+> * LGPL
+> * No license
+
+This includes the license of any code that your dependency depends on! If you are ever
+unsure, feel free to ask us on discord.
+
+
+### (Optional) Fork the dependency
+
+When you want to use the depdendency, you should ask yoursef: "Can I simply call the
+functions/classes in the dependency in *my* code, or do I actually need to modify the
+source code of the original files"? If the answer is "I need to modify the origianl
+files", then you will need to fork the dependency so that your modifications are clear.
+
+> **Note**
+> If you have never forked in git before, a fork is basically a copy of a library that
+> maintains the original's history. This means that any changes you commit in your fork
+> will show up as changes against the original. We can even update these changes as the
+> original changes!
+
+To create a fork in github, simply go to the original library and click this button:
+![](https://docs.github.com/assets/cb-28613/images/help/repository/fork_button.png)
+Now clone your repo to your computer. From this point forward, any changes that you
+make while working on your fork will have their git history preserved and linked to the
```

Comment:
```suggestion
make while working on your fork will have their Git history preserved and linked to the
```

Reply:
![](https://c.tenor.com/QA_IqSKoWTcAAAAC/the-rock.gif)

Reply:
I don't think we need to capitalize all uses of git throughout the codebase - I pretty much only see git used lowercase.